common/djangoapps/student/views.py

```diff
@@ -1718,7 +1722,7 @@ def create_account_with_params(request, params):
     # Perform operations within a transaction that are critical to account creation
     with transaction.atomic():
         # first, create the account
-        (user, profile, registration) = _do_create_account(form, custom_form)
+        (user, profile, registration) = _do_create_account(form, custom_form, site=getattr(request, 'site', None))
```

Comment:
The headaches are admittedly minor. You have to worry about testing both cases and you cannot make any guarantees down the road. It would be ideal if we established the pattern now that all new accounts will have a link To a site.